int32 start
int32 goal
---
int32[] path
